teps/0150-displayname-in-matrix.md

```diff
@@ -0,0 +1,304 @@
+---
+
+# TEP-0150: displayName in Matrix
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Matrix feature of Tekton Pipelines allows the `pipeline` author to specify multiple concurrent execution of the
+same `pipelineTask`. The same `pipelineTask` is executed in parallel based on the number of input combination of
+parameters. 
+
+Matrix supports two different types of specifications including [implicit](https://github.com/tektoncd/pipeline/blob/main/docs/matrix.md#generating-combinations)
+and [explicit](https://github.com/tektoncd/pipeline/blob/main/docs/matrix.md#explicit-combinations) combinations of
+parameters.
+
+- With implicit combinations, a `pipelineTask` is defined with a list of parameters and one or more values for each
+  parameter. Tekton controller generates an exhaustive list of combinations based on the specified parameters. Based on
+  the generated list of combinations, a `pipelineTask` is fanned out and executed with each combination. The number of
+  `taskRun`s in this case is not predictable and depends on the runtime configuration.
+
+```yaml
+- name: platforms-and-browsers
+  matrix:
+    params:
+      - name: platform
+        value: $(params.platforms[*])
+      - name: browser
+        value: $(params.browsers[*])
+```
+
+- With explicit combinations, a `pipelineTask` is defined with a list of combinations and Tekton controller fans out the
+  task based on the number of combinations. Each running instance receives unique combination of input parameters. The
+  pipeline author can easily predict the number of parallel instances in advance based on this specification.
+
+```yaml
+- name: platforms-and-browsers
+  matrix:
+      include:
+        - params:
+            - name: platform
+              value: "linux"
+            - name: browser
+              value: "chrome"
+        - params:
+            - name: platform
+              value: "mac"
+            - name: browser
+              value: "safari"
+```
+
+The Tekton Dashboard lists all the instances created for a `pipelineTask` with `matrix` as expected but all have the same
+`pipelineTask` name.
+
+![PipelineRun With Matrix](images/0150-pipelinerun-with-matrix.png)
+
+The users have to click on each `pipelineTask` tab to find the combination of values used for each instance.
+
+## Motivation
+
+The `matrix` functionality in a `pipeline` provides an ability to execute the same `task` with variety of combinations but
+distinguishing each instance of a `matrix` is not easy through Tekton Dashboard. The motivation of this proposal is to
+design an API such that the pipeline authors can specify the distinguishing parameter for each `matrix` instance. This
+new API can be utilized by the Tekton Dashboard to display unique names.
```

Comment:
This issue is not specific to the Tekton Dashboard, all clients likely face the same problem.

Similar below, the solution is not designed specifically for the Tekton Dashboard but would benefit all clients.